### PR TITLE
AAP-35261: ModelPipelines: Remove duplicate empty implementations

### PR DESCRIPTION
**Pull request by manstis**
Jira Issue: https://issues.redhat.com/browse/AAP-35261

## Description
This is a refactoring to simplify existing code; removing duplications.

## Testing
N/A. Unit test coverage suffices.

### Steps to test
N/A. Unit test coverage suffices.

### Scenarios tested
N/A. Unit test coverage suffices.

## Production deployment
- [x] This code change is ready for production on its own
- [ ] This code change requires the following considerations before going to production:

### CODE REVIEW

File: ansible_ai_connect/ai/api/model_pipelines/registry.py [hunk 1]
```diff
@@ -89,6 +89,13 @@
         ModelPipelinePlaybookGeneration: None,
         ModelPipelinePlaybookExplanation: None,
     },
+    "nop": {
+        MetaData: None,
+        ModelPipelineCompletions: None,
+        ModelPipelineContentMatch: None,
+        ModelPipelinePlaybookGeneration: None,
+        ModelPipelinePlaybookExplanation: None,
+    },
```
Comment (hasys) [inline ansible_ai_connect/ai/api/model_pipelines/registry.py:98]: Since we already refactoring the code around, maybe we can replace all of the repetitions with

```python
import copy

EMPTY_PIPE = {
         MetaData: None,
         ModelPipelineCompletions: None,
         ModelPipelineContentMatch: None,
         ModelPipelinePlaybookGeneration: None,
         ModelPipelinePlaybookExplanation: None,
     }
PIPELINES = {
    "grpc": copy.deepcopy(EMPTY_PIPE),
    "http": copy.deepcopy(EMPTY_PIPE),
    [...]
    "nop": copy.deepcopy(EMPTY_PIPE)
}
```

So next time we need to add/change a pipeline we can do it in one place. And it looks more readable as well.

Comment (hasys) [reply ansible_ai_connect/ai/api/model_pipelines/registry.py:98]: Also, we can combine with base class, to reduce amount of copy paste even further:
```python
from copy import deepcopy
from ansible_ai_connect.main.settings.base import t_model_mesh_api_type

EMPTY_PIPE = {
         MetaData: None,
         ModelPipelineCompletions: None,
         ModelPipelineContentMatch: None,
         ModelPipelinePlaybookGeneration: None,
         ModelPipelinePlaybookExplanation: None,
     }
PIPELINES = {}
for model_mesh in t_model_mesh_api_type:
    PIPELINES[model_mesh] = deepcopy(EMPTY_PIPE)
```

**Review by hasys — COMMENTED** [review]
Hi @manstis, looks great to me, thank you!

Just 2c to remove duplication even better. Also, thinking about adding tests for all removed empty implementations, just to be sure they are empty in the future.

**Comment by manstis** [comment]
@hasys More tests and other updates you proposed made. 

Please re-review when you have a chance.

**Review by hasys — APPROVED** [review]
Looks good, great job, thank you @manstis!
